transfer/rsync/client.go

```diff
@@ -0,0 +1,431 @@
+	rsyncCommand = append(rsyncCommand, fmt.Sprintf("/mnt/%s/%s", pvc.Claim().Namespace, pvc.LabelSafeName()))
+	rsyncCommand = append(rsyncCommand,
+		fmt.Sprintf("rsync://%s@%s/%s --port %d",
+			tc.username,
+			tc.Transport().Hostname(),
+			pvc.LabelSafeName(), tc.Transport().ListenPort()))
+	rsyncCommandBashScript := fmt.Sprintf(
+		"trap \"touch /usr/share/rsync/rsync-client-container-done\" EXIT SIGINT SIGTERM; timeout=120; SECONDS=0; while [ $SECONDS -lt $timeout ]; do nc -z localhost %d; rc=$?; if [ $rc -eq 0 ]; then %s; rc=$?; break; fi; done; exit $rc;",
+		tc.Transport().ListenPort(),
+		strings.Join(rsyncCommand, " "))
+	rsyncContainerCommand := []string{
+		"/bin/bash",
+		"-c",
+		rsyncCommandBashScript,
+	}
+	return rsyncContainerCommand
+}
+
+// customizeTransportClientContainers customizes transport's client containers for specific rsync communication
+func customizeTransportClientContainers(transportClient transport.Transport) error {
+	switch transportClient.Type() {
+	case stunnel.TransportTypeStunnel:
+		var stunnelContainer *corev1.Container
+		for i := range transportClient.Containers() {
+			c := &transportClient.Containers()[i]
+			if c.Name == stunnel.Container {
+				stunnelContainer = c
+			}
+		}
+		if stunnelContainer == nil {
+			return fmt.Errorf("couldnt find container named %s in rsync client pod", stunnel.Container)
+		}
+		stunnelContainer.Command = []string{
+			"/bin/bash",
+			"-c",
+			`/bin/stunnel /etc/stunnel/stunnel.conf
+while true
+do test -f /usr/share/rsync/rsync-client-container-done
+if [ $? -eq 0 ]
+then
+break
+fi
+done
+exit 0`,
+		}
+		stunnelContainer.VolumeMounts = append(
+			stunnelContainer.VolumeMounts,
+			corev1.VolumeMount{
+				Name:      "rsync-communication",
+				MountPath: "/usr/share/rsync",
+			})
+	}
+	return nil
+}
+
+func (tc *client) reconcileServiceAccount(ctx context.Context, c ctrlclient.Client, namespace string) error {
+	sa := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s", rsyncServiceAccount, tc.nameSuffix),
+			Namespace: namespace,
+		},
+	}
+	_, err := ctrlutil.CreateOrUpdate(ctx, c, sa, func() error {
+		sa.Labels = tc.labels
+		sa.OwnerReferences = tc.ownerRefs
+		return nil
+	})
+	return err
+}
+
+func (tc *client) reconcileRole(ctx context.Context, c ctrlclient.Client, namespace string) error {
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s", rsyncRole, tc.nameSuffix),
+			Namespace: namespace,
+		},
+	}
+	_, err := ctrlutil.CreateOrUpdate(ctx, c, role, func() error {
+		role.OwnerReferences = tc.ownerRefs
+		role.Labels = tc.labels
+		role.Rules = []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"security.openshift.io"},
+				Resources: []string{"securitycontextconstraints"},
+				// Must match the name of the SCC that is deployed w/ the operator
+				// config/openshift/mover_scc.yaml
+				ResourceNames: []string{DefaultSCCName},
```

Comment:
We're going to need to find a way to deal with this. Looks like it's a hardcoded name, but that may not work well for library users.